src/main/python/RemoteSwingLibrary.py

```diff
@@ -69,18 +69,25 @@ def set_received_to_old(self):
 
 REMOTE_AGENTS_LIST = AgentList()
 
-class SimpleServer(SocketServer.BaseRequestHandler):
+class SimpleServer(SocketServer.StreamRequestHandler):
 
     def handle(self):
-        data = ''.join(iter(self.read_socket, ''))
-        port, name = data.decode().split(':', 1)
-        address = ':'.join([self.client_address[0], port])
-        logger.debug('Registered java remoteswinglibrary agent "%s" at %s' % \
-              (name, address))
-        REMOTE_AGENTS_LIST.append(address, name)
-
-    def read_socket(self):
-        return self.request.recv(1)
+        while True:
```

Comment:
this while loop should not be needed, we call `serve_forever`, so `handle` should be called multiple times to handle all messages, but somehow only first message is processed without this loop. @jussimalinen, do you have any idea what is happening here?